### DISCUSSION TITLE
Verbose outputs for the "attempt to index a nil value" error

### DISCUSSION THREAD

**Discussion by kotobuki** [zepinglee/citeproc-lua#70] [Ideas]
Hi,

So far, citeproc-lua has worked fine in my environment (TeX Live 2024 on macOS Sonoma 14.5). It is a very useful tool, and I sincerely appreciate it. However, after recently updating the JSON file (generated by Zotero and Better BibTex), the following error occurs:

```
%xelatex main.tex
...
% citeproc-lua main.aux
This is citeproc-lua, Version 0.5.1
The top-level auxiliary file: main.aux
Global style file: chicago-fullnote-bibliography-short-title-subsequent-ja.csl
attempt to index a nil value
```

I found the error's cause was the following part in the main.tex file (missing a page number or a page range).

`\cite[page = ]{auth2024title}`

Of course, it was a stupid mistake by me, but It would be nice if there were an option to get verbose outputs (e.g., with the line number) to narrow down the problem.

Thank you very much in advance.

**Comment by zepinglee** [comment]
Thanks for your feedback! I've just fixed this error in 709dc18. Now the package raises a warning in this case and `citeproc-lua` doesn't break with such error (see the following test result with two passes of `pdflatex`).

https://github.com/zepinglee/citeproc-lua/blob/709dc1885b173ec28fe4648eb4f56dc6a3dea22a/tests/latex/pdftex-2/issue-70.tlg#L6-L12

I didn't consider much about user-friendly error or warning messages when I wrote this package. I would appreciate more cases and enhancements like this.
